flutter_local_notifications/docs/usage.md

```diff
@@ -0,0 +1,269 @@
+
+```dart
+await plugin.zonedSchedule(
+  0, "Your event is now!", "This is a body",
+  tz.TZDateTime.now(tz.local).add(const Duration(seconds: 5)),
+  NotificationDetails(...),
+  androidScheduleMode: AndroidScheduleMode.exactAllowWhileIdle,
+  payload: "goto-event_EVENT_ID",
+);
+```
+
+By default, notifications shown with this function will not repeat unless `matchDateTimeComponents` is provided, in which case the [`DateTimeComponents`](https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/DateTimeComponents.html) will specify how to repeat. For example, passing `DateTimeComponents.dayOfWeekAndTime` will make the notification repeat once a week, during the given date time. Where possible, though, try to use [`periodicallyShowWithDuration()`](https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/FlutterLocalNotificationsPlugin/periodicallyShowWithDuration.html) (see below).
+
+### Periodically show a notification
+
+> [!Note]
+>
+> Not supported on Linux, Windows, or the Web
+
+To periodically show a notification, use [`periodicallyShowWithDuration()`](https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/FlutterLocalNotificationsPlugin/periodicallyShow.html). It takes the same arguments as `show()`, but also accepts a `repeatIntervalDuration` specifying how far apart these notifications should be and an `androidScheduleMode` as explained above. For example:
+
+```dart
+await plugin.periodicallyShowWithDuration(
+  0, "Have you moved today?", "Exercise is important!",
+  Duration(hours: 1),
+  NotificationDetails(...),
+  androidScheduleMode: AndroidScheduleMode.inexact,
+  payload: "yes-they-moved",
+)
+```
+
+### Notification actions
+
+While many details differ between platforms, all platforms support the concept of notification actions. See your platform's usage guide for exact details, but the general idea is allowing a user to select an action within your notification itself, triggering a response from your application without requiring the user to open your app and select it themselves. In some platforms and circumstances, this action may be triggered entirely in the background, allowing your user to stay in their current app.
+
+For example, imagine a messaging app that just got a message from another user. Your notification may include a text action, allowing the user to send a response directly within the notification. Again, the exact details here differ by platform.
+
+Actions affect how your app handles responses. The foreground handler, background handler, and `getNotificationAppLaunchDetails()` method all provide you with a `NotificationResponse` object, which will contain a non-null `actionId` if an action was used, with a non-null `input` (or `data`) if a text field was used. For example, a notification can have a payload `new-message_CHAT_ID` and an `actionId` of `mark-read`.
```

Comment:
See comment on foreground/background handler that has implications on this section. Once addressed and renamed, can add a link to the sections on said handlers too